File: week2-math/problem07-소수찾기/heetae.java
/**
문제: 소수찾기
접근 방식: 1부터 소수판별법을 사용 -> 이게 빠른줄 알았는데 에라토스테네스의 체가 더빠르다고 함
시간복잡도: O(n√n)
*/
class Solution {
    public int solution(int n) {
        int answer = 0;
        for(int i =1;i<=n;i++){
            if(isPrime(i)) answer++;
        }
        return answer;
    }
    boolean isPrime(int n){
        if(n<=1) return false;
        if(n==2 || n==3) return true;
        if(n%2==0 || n%3==0) return false;

        for(int i=5;i*i<=n;i+=6){
            if(n%i==0 || n%(i+2) == 0) return false;
        }

        return true;
    }
}

File: week1-array-string/problem12-문자열압축/heetae.py
# 문제: 문자열 압축
# 접근 방식: 1부터 문자열 길이의 절반까지 압축 단위를 변경하며, 연속한 동일 문자열의 개수를 세어 가장 짧은 압축 길이를 구한다.
# 하지만 파이썬 문자 자르기가 익숙하지 않아 틀림,,,
# 시간복잡도: O(n^2) (n은 문자열의 길이)

def solution(s):
    answer = len(s)

    for size in range(1, len(s) // 2 + 1):

        compressed = ""
        prev = s[:size]
        count = 1

        for i in range(size, len(s), size):
            chunk = s[i:i+size]

            if prev == chunk:
                count += 1

            else:
                if count == 1:
                    compressed += prev
                else:
                    compressed += str(count) + prev

                prev = chunk
                count = 1

        # 마지막 덩어리 처리
        if count == 1:
            compressed += prev
        else:
            compressed += str(count) + prev

        # 최소 길이 갱신
        answer = min(answer, len(compressed))

    return answer
